src/DynamoCoreWpf/ViewModels/Search/BrowserItemViewModel.cs

```diff
@@ -152,6 +153,30 @@ public string Assembly
             }
         }
 
+        public ElementTypes ElementType
+        {
+            get
+            {
+                if (entries.Any(entry => entry.ElementType.HasFlag(ElementTypes.BuiltIn)) ||
+                    subCategories.Any(subCat => subCat.ElementType.HasFlag(ElementTypes.BuiltIn)))
+                    return ElementTypes.BuiltIn;
+
+                if (entries.All(entry => entry.ElementType.HasFlag(ElementTypes.Packaged)) &&
+                    subCategories.All(subCat => subCat.ElementType.HasFlag(ElementTypes.Packaged)))
+                    return ElementTypes.Packaged;
+
+                if (entries.All(entry => entry.ElementType.HasFlag(ElementTypes.ZeroTouch)) &&
+                    subCategories.All(subCat => subCat.ElementType.HasFlag(ElementTypes.ZeroTouch)))
+                    return ElementTypes.ZeroTouch;
+
+                if (entries.All(entry => entry.ElementType.HasFlag(ElementTypes.CustomNode)) &&
+                    subCategories.All(subCat => subCat.ElementType.HasFlag(ElementTypes.CustomNode)))
+                    return ElementTypes.CustomNode;
+
+                return ElementTypes.None;
+            }
+        }
+
```

Comment:
I'd prefer this to be done less frequently, I'm imagining this `ElementType` to be called each time the UI needs the information to display an icon, then that might be too many evaluation on these collections `entries` and `subCategories`. If we can do them when these two collection changed, then it's one-time evaluation:

```csharp
private void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
{
    // Perform the "entries.Any" and "subCategories.Any" calls here, then set "ElementType"...
    if (entries.Any ... && subCategories.Any ...)
        ElementType = ElementTypes.BuiltIn;
}

public ElementTypes ElementType { get; private set; }
```

If you can do a count and see if we get more calls for `ElementType.get` or `OnCollectionChanged`, then decide which is the best way to go about doing it. Please let me know.